protocol/interop-topology.md

```diff
@@ -0,0 +1,145 @@
+to validate cross chain messages rather than its local index. This could be accomplished by adding
+new RPC endpoints that do this or could be done with runtime config. When `op-supervisor` runs in this
+mode, it is a "light mode" that only supports `supervisor_validateMessagesV2` and `supervisor_validateAccessList`
+(potentially a subset of their behavior). This would give us a form of "client diversity" with respect
+to validating cross chain messages. This is a low lift way to reduce the likelihood of a forged initiating
+message. A forged initiating message would be tricking the caller into believing that an initiating
+message exists when it actually doesn't, meaning that it could be possible for an invalid executing
+message to finalize.
+
+TODO: feedback to understand what capabilities are possible
+
+### `proxyd`
+
+We update `proxyd` to validate interop messages on cloud ingress. It should check against both the indexed
+backend of `op-supervisor` as well as the alternative backend.
+
+### Sentry Node Mempool Ingress
+
+We update the EL clients to validate interop transactions on ingress to the mempool. This should be a different
+instance of `op-supervisor` than the one that is used by `proxyd` to reduce the likelihood of a nondeterministic
+bug within `op-supervisor`.
+
+### Sentry Node + Sequencer Mempool on Interval
+
+We update the EL clients to validate interop transactions on an interval in the mempool. Generally the mempool
+will revalidate all transactions on each new block, but for an L2 that has 1-2s blocktime, that is quite often.
+It may be the case that we could get away with a batch RPC request every 1-2s, but generally we should not do
+`n` RPC requests on each block where `n` is the number of transactions that include statically declared executing
+messages.
+
+### Block Building
+
+Lets say that it takes 100ms for the transaction to be checked at `proxyd`, checked at the mempool of the sentry node,
+forwarded to the sequencer and pulled into the block builder. The chances of the status of an initiating message
+going from existing to not existing during that timeframe is extremely small. Even if we did check at the block builder,
+it doesn't capture the case of a future unsafe chain reorg happening that causes the message to become invalid.
+Because it is most likely that the remote unsafe reorg comes after the local block is sealed, there is no real
+reason to block the hot path of the chain with the remote lookups.
+
+## Resource Usage
+
+<!-- What is the resource usage of the proposed solution?
+Does it consume a large amount of computational resources or time? -->
+
+Doing a remote RPC request is always going to be an order of magnitude slower than doing a local lookup.
+Therefore we want to ensure that we can parallelize our remote lookups as much as possible. Block building
+is inherently a single threaded process given that the ordering of the transactions is very important.
```

Comment:
Because every transaction has a statically defined Access List, *unless* the transactions being considered for building are interdependent (which is an unlikely use-case), interop transactions *can* indeed be batch submitted.

Say the builder started building, and before it did it looked over the first "100" Tx in its prioritized list. It could locally ilter out all the access list items across all the transactions, and then submit it to the Supervisor as one giant `CheckAccessList` call.

This would allow us to have very time-relevant checks while still not individually checking every Tx while building, which I agree could be prohibitive.

Reply:
One issue with this -- if we batched up X Tx's AccessLists and sent it to one big CheckAccessList call, if it fails we won't immediately know *which* transaction fails. That might be a moment where we can break it down and move slower because we know *something* is going to be invalid, and then in the happy-path we can still just use the big batches.

Reply:
If you do a batch rpc call then you should be able to get a batch rpc response and know which responses correspond to which requests by matching up indices

Reply:
Fair, it could be a batch of CheckAccessList calls. I was imagining that we could compile them into one *massive* Access List and then just make a single (non-batched) call, which loses the Tx resolution. But just baching the calls make sense.